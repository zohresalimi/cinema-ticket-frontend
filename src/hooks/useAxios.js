import { useEffect, useReducer, useMemo, useCallback } from "react";
import axios from "axios";
import qs from "qs";

const baseUrl = process.env.REACT_APP_BASE_URL;
const actions = {
  REQUEST_START: "REQUEST_START",
  REQUEST_END: "REQUEST_END",
};

async function fetchData({ path, ...config }, dispatch) {
  try {
    dispatch({ type: actions.REQUEST_START });
    const result = await axios({
      ...config,
      url: `${baseUrl}${path}`,
      paramsSerializer: (params) => {
        return qs.stringify(params);
      },
    });
    dispatch({ type: actions.REQUEST_END, payload: result.data });
  } catch (err) {
    dispatch({
      type: actions.REQUEST_END,
      payload: err,
      error: true,
    });
  }
}

const useAxios = (path, config) => {
  function reducer(state, action) {
    switch (action.type) {
      case actions.REQUEST_START:
        return {
          ...state,
          response: null,
          loading: true,
          error: null,
        };
      case actions.REQUEST_END:
        return {
          ...state,
          loading: false,
          ...(action.error ? {} : { data: action.payload.data }),
          [action.error ? "error" : "response"]: action.payload,
        };
      default:
        return state;
    }
  }

  function createInitialState({ manual }) {
    return {
      loading: !manual,
      error: null,
      response: null,
    };
  }

  const options = useMemo(
    () => ({
      data: {},
      method: "get",
      manual: false,
      path,
      ...config,
    }),
    [config, path]
  );
  const [state, dispatch] = useReducer(reducer, createInitialState(options));

  useEffect(() => {
    if (!options.manual) {
      fetchData(options, dispatch);
    }
  }, [options, dispatch, path]);

  const refetch = useCallback(
    (payload) => {
      return fetchData({ ...options, ...payload }, dispatch);
    },
    [options, dispatch]
  );

  return [state, refetch];
};

export default useAxios;
