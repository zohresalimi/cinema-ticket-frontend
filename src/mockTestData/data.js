import React, { useReducer } from "react";
import { Router } from "@reach/router";
import AppContext from "../store/context";
import Theme from "../Styles/Theme";
import reducer from "../store/reducer";

export const getTestStore = () => {
  return {
    user: {
      name: "",
      email: "",
    },
    movies: {
      children: [
        {
          _id: "5fa71823adc463397a50c501",
          actors: ["Holland Roden", "Ronen Rubinstein", "Keegan Allen"],
          age: "7",
          coverImage:
            "https://catalog.cinema-api.com/cf/images/ncg-images/68bfeef55ada4e7ab0606db2af0f40ad.jpg?width=240&version=EA2E80EE161E5DA161C6E26487054C9F&format=webp",
          description:
            "Poppy and Kvist embark on an adventure that will take them far beyond",
          director: "aaron woodley",
          duration: "1:31",
          genre: ["family", "children"],
          images: [],
          largeImage:
            "https://catalog.cinema-api.com/cf/images/ncg-images/79276cead82341d2a004f8e8f99a88e9.jpg?width=1920&version=08F0C83177D59FB94C4C809F63F67E2A&format=webp",
          name: "trolls 2: world tour",
          originalTitle: "swedish",
          premiere: "2020-10-03T15:18:52.722Z",
          rooms: ["5fa653d0a7f1201999dd1ac0"],
          trailerUrl: "https://youtu.be/P9A6hOg9QQ0",
        },
      ],
      premiered: [
        {
          _id: "5fa71823adc463397a50c502",
          actors: ["Holland Roden", "Ronen Rubinstein", "Keegan Allen"],
          age: "7",
          coverImage:
            "https://catalog.cinema-api.com/cf/images/ncg-images/68bfeef55ada4e7ab0606db2af0f40ad.jpg?width=240&version=EA2E80EE161E5DA161C6E26487054C9F&format=webp",
          description:
            "Poppy and Kvist embark on an adventure that will take them far beyond",
          director: "aaron woodley",
          duration: "1:31",
          genre: ["family", "children"],
          images: [],
          largeImage:
            "https://catalog.cinema-api.com/cf/images/ncg-images/79276cead82341d2a004f8e8f99a88e9.jpg?width=1920&version=08F0C83177D59FB94C4C809F63F67E2A&format=webp",
          name: "follow me",
          originalTitle: "swedish",
          premiere: "2020-10-05T15:18:52.722Z",
          rooms: ["5fa653c0643c06586bba7d78"],
          trailerUrl: "https://youtu.be/P9A6hOg9QQ0",
        },
      ],
      upcoming: [
        {
          _id: "5fa71823adc463397a50c503",
          actors: ["Holland Roden", "Ronen Rubinstein", "Keegan Allen"],
          age: "7",
          coverImage:
            "https://catalog.cinema-api.com/cf/images/ncg-images/68bfeef55ada4e7ab0606db2af0f40ad.jpg?width=240&version=EA2E80EE161E5DA161C6E26487054C9F&format=webp",
          description:
            "Poppy and Kvist embark on an adventure that will take them far beyond",
          director: "aaron woodley",
          duration: "1:31",
          genre: ["family", "children"],
          images: [],
          largeImage:
            "https://catalog.cinema-api.com/cf/images/ncg-images/79276cead82341d2a004f8e8f99a88e9.jpg?width=1920&version=08F0C83177D59FB94C4C809F63F67E2A&format=webp",
          name: "honest thief",
          originalTitle: "swedish",
          premiere: "2021-01-03T15:18:52.722Z",
          rooms: ["5fa653c0643c06586bba7d78"],
          trailerUrl: "https://youtu.be/P9A6hOg9QQ0",
        },
      ],
      allMovies: {
        "5fa71823adc463397a50c501": "children",
        "5fa71823adc463397a50c502": "premiered",
        "5fa71823adc463397a50c503": "upcoming",
      },
    },
    cinemas: [
      {
        _id: "5fa64fe234fbbf9945f3a3ce",
        name: "filmstaden kista",
        purchaseEndTime: "21:30",
        purchaseStartTime: "8:00",
        rooms: ["5fa653c0643c06586bba7d78", "5fa653d0a7f1201999dd1ac0"],
      },
    ],
    ticket: {
      movie: {},
      status: false,
      quantity: 0,
      price: 0,
      seatNumbers: new Map(),
      user: {},
      showing: {},
    },
    showings: [
      {
        _id: "5fbe9bd31f58462b5c759b5c",
        capacity: 4,
        cinema: "5fa64fe234fbbf9945f3a3ce",
        endTime: null,
        movie: {
          _id: "5fa71823adc463397a50c502",
          actors: ["Holland Roden", "Ronen Rubinstein", "Keegan Allen"],
          age: "7",
          description:
            "Poppy and Kvist embark on an adventure that will take them far beyond",
          director: "aaron woodley",
          duration: "1:31",
          genre: ["family", "children"],
          images: [],
          name: "follow me",
          originalTitle: "swedish",
          premiere: "2020-10-05T15:18:52.722Z",
          rooms: ["5fa653c0643c06586bba7d78"],
          trailerUrl: "https://youtu.be/P9A6hOg9QQ0",
        },
        price: 100,
        room: { _id: "5fa653c0643c06586bba7d78" },

        seats: [
          [
            { seatNumber: 1, taken: "available" },
            { seatNumber: 2, taken: "available" },
          ],
          [
            { seatNumber: 1, taken: "available" },
            { seatNumber: 2, taken: "available" },
          ],
        ],
        startTime: "2020-11-13T16:00:00.000Z",
      },
    ],
  };
};

export const WithProvider = (props) => {
  const [state, dispatch] = useReducer(reducer, getTestStore());

  return (
    <Theme>
      <AppContext.Provider
        value={{
          state: props.defaultValue || state,
          dispatch: props.mockDispatch || dispatch,
        }}
      >
        <Router>{props.children}</Router>
      </AppContext.Provider>
    </Theme>
  );
};
