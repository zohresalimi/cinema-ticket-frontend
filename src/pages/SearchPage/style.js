import styled from "styled-components";

export const Div = styled.div`
  flex-direction: row;
  justify-content: inherit;
  align-items: center;
  display: flex;
  height: 100%;
`;

export const TopMenuContainer = styled.div`
  font-family: ${(props) => props.theme.fontFamily.sf};
  display: flex;
  align-items: center;
  width: 100%;
  height: 4rem;
  background-color: ${(props) => props.theme.colors.red};
  color: ${(props) => props.theme.colors.white};
  .searchBar {
    flex-direction: row;
    justify-content: flex-end;
    align-items: center;
    display: flex;
    .searchIcon {
    }
  }
`;

export const SearchInput = styled.div`
  width: 100%;
  display: flex;
  position: relative;
  margin-top: 20px;
  svg{
      position: absolute;
      top: 34%;
    left: 13px;
    &.spiner{
      position: absolute;
      top: 34%;
    left: auto;
    right: 50px;
  }
  }

  input {
    width: 95%;
    border-radius: 1.5rem;
    padding: 10px;
    background-color: ${(props) => props.theme.colors.black};
    border: 1px solid #fff;
    font-family: ${(props) => props.theme.fontFamily.sf};
    font-size: ${(props) => props.theme.fontSizes.content};
    color: ${(props) => props.theme.colors.white};
    text-indent: 40px;
    ::-webkit-input-placeholder {
      color: color: ${(props) => props.theme.colors.lightGray};
    }
    :-moz-placeholder {
      color: color: ${(props) => props.theme.colors.lightGray};
      opacity: 1;
    }
    ::-moz-placeholder {
      color: color: ${(props) => props.theme.colors.lightGray};
      opacity: 1;
    }
    :-ms-input-placeholder {
      color: color: ${(props) => props.theme.colors.lightGray};
    }
    ::-ms-input-placeholder {
      color: color: ${(props) => props.theme.colors.lightGray};
    }

    ::placeholder {
      color: color: ${(props) => props.theme.colors.lightGray};
    }
    &:focus {
    outline: none !important;
    border:1px solid red;
    border-radius: 1.5rem;
    box-shadow: 0 0 10px #719ECE;
}
  }
`;
