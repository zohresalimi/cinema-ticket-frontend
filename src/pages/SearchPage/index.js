/* eslint-disable react-hooks/exhaustive-deps */
import React, { useState, useEffect, useContext } from "react";
import { Link } from "@reach/router";
import { useTranslation } from "react-i18next";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSearch, faSpinner } from "@fortawesome/free-solid-svg-icons";
import TopBar from "../../components/TopBar";
import useAxios from "../../hooks/useAxios";

import { SearchInput, Wrapper } from "./style";
import { Container, Row } from "../../Styles/StyleComponents";
import AppContext from "../../store/context";

function SearchPage() {
  const { state } = useContext(AppContext);
  const { allMovies } = state.movies;
  const { t } = useTranslation();
  const [input, setInput] = useState({ value: "" });

  const [
    { response: searchResponse, loading: searchLoading },
    searchResult,
  ] = useAxios("/api/v1/movies/search", {
    manual: true,
    params: {
      query: input.value,
    },
  });

  useEffect(() => {
    if (input.value) {
      searchResult();
    }
  }, [input.value]);

  return (
    <div>
      <TopBar />
      <Wrapper>
        <Container>
          <Row>
            <SearchInput>
              <input
                name="search-input"
                type="text"
                autoComplete="off"
                data-testid="search-input"
                placeholder={t("search by movie name!")}
                onChange={(e) => setInput({ value: e.target.value })}
              />
              <FontAwesomeIcon icon={faSearch} className="searchIcon" />
            </SearchInput>
            {searchLoading && (
              <div className="spiner-wraper">
                <FontAwesomeIcon icon={faSpinner} className="spiner" spin />
              </div>
            )}
            <ul id="movie-List">
              {searchResponse && !searchResponse.response.length && (
                <div className="no-result-message">
                  <p>
                    {t(
                      "Could not find movies with this search term, give me more characters."
                    )}
                  </p>
                </div>
              )}
              {searchResponse && searchResponse.response.length
                ? searchResponse.response.map((movie) => {
                    return (
                      <li key={movie._id}>
                        <Link
                          to={`/${
                            allMovies[movie._id] || "search"
                          }/movie-detail/${movie._id}`}
                        >
                          <div className="image-wrapper">
                            <img src={movie.coverImage} alt="" />
                          </div>
                          <div>
                            <p>{movie.name}</p>
                            <p>{movie.genre}</p>
                          </div>
                          <div className="right-side">{t("show")}</div>
                        </Link>
                      </li>
                    );
                  })
                : ""}
            </ul>
          </Row>
        </Container>
      </Wrapper>
    </div>
  );
}

export default SearchPage;
