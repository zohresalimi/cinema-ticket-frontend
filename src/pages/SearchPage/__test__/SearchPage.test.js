import * as React from "react";

import { act, render, fireEvent, waitFor } from "@testing-library/react";
import { when } from "jest-when";
import "@testing-library/jest-dom/extend-expect";

import * as hooks from "../../../hooks/useAxios";
import { getTestStore, WithProvider } from "../../../mockTestData/data";
import SearchPage from "../";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({
    t: (k) => k,
    i18n: {
      language: "sv",
      changeLanguage: () => new Promise(() => {}),
    },
  }),
}));

jest.mock("../../../components/TopBar", () => () => (
  <div>TopBar Component</div>
));

async function renderWrapper(props) {
  let component;

  act(() => {
    component = render(
      <WithProvider>
        <SearchPage {...props} path="/" />
      </WithProvider>
    );
  });

  return component;
}

describe("Search Page Component Testing", () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  test("take snapshot", async () => {
    const { container } = await renderWrapper();
    expect(container.firstChild).toMatchSnapshot();
  });

  test("should call to search api", async () => {
    const mockUseAxios = jest.fn();
    when(mockUseAxios)
      .calledWith("/api/v1/movies/search", {
        manual: true,
        params: {
          query: "",
        },
      })
      .mockReturnValue([
        {
          response: null,
          loading: true,
        },
        jest.fn(),
      ]);

    jest.spyOn(hooks, "default").mockImplementation(mockUseAxios);
    const { container } = await renderWrapper();
    expect(container.firstChild).toMatchSnapshot();
  });

  test("should render message when api returns empty list", async () => {
    const mockUseAxios = jest.fn();
    when(mockUseAxios)
      .calledWith("/api/v1/movies/search", {
        manual: true,
        params: {
          query: "",
        },
      })
      .mockReturnValue([
        {
          response: {
            response: [],
          },
          loading: false,
        },
        jest.fn(),
      ]);

    jest.spyOn(hooks, "default").mockImplementation(mockUseAxios);
    const { container } = await renderWrapper();
    expect(container.firstChild).toMatchSnapshot();
  });

  test("should render items when api returns result", async () => {
    const mockUseAxios = jest.fn();
    when(mockUseAxios)
      .calledWith("/api/v1/movies/search", {
        manual: true,
        params: {
          query: "",
        },
      })
      .mockReturnValue([
        {
          response: {
            response: getTestStore().movies.premiered,
          },
          loading: false,
        },
        jest.fn(),
      ]);
    when(mockUseAxios)
      .calledWith("/api/v1/movies/search", {
        manual: true,
        params: {
          query: "1",
        },
      })
      .mockReturnValue([
        {
          response: {
            response: getTestStore().movies.premiered,
          },
          loading: false,
        },
        jest.fn(),
      ]);

    jest.spyOn(hooks, "default").mockImplementation(mockUseAxios);
    const { container, getByTestId } = await renderWrapper();
    const searchInput = getByTestId("search-input");
    fireEvent.change(searchInput, {
      target: {
        value: "1",
      },
    });
    expect(mockUseAxios).toHaveBeenCalledTimes(2);
    expect(container.firstChild).toMatchSnapshot();
  });
});
