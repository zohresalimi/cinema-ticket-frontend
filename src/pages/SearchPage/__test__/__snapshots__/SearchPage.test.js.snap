// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Page Component Testing should call to search api 1`] = `
<div
  style="outline: none;"
  tabindex="-1"
>
  <div>
    <div>
      TopBar Component
    </div>
    <div
      class="style__Wrapper-sc-184j6m9-0 iDVMYk"
    >
      <div
        class="StyleComponents__Container-sc-11erdxt-0 fFgwWO"
      >
        <div
          class="StyleComponents__Row-sc-11erdxt-1 jhoSVF"
        >
          <div
            class="style__SearchInput-sc-184j6m9-1 bWXmHs"
          >
            <input
              autocomplete="off"
              data-testid="search-input"
              name="search-input"
              placeholder="search by movie name!"
              type="text"
            />
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-search fa-w-16 searchIcon"
              data-icon="search"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                fill="currentColor"
              />
            </svg>
          </div>
          <div
            class="spiner-wraper"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-spinner fa-w-16 fa-spin spiner"
              data-icon="spinner"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
                fill="currentColor"
              />
            </svg>
          </div>
          <ul
            id="movie-List"
          >
            
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Search Page Component Testing should render items when api returns result 1`] = `
<div
  style="outline: none;"
  tabindex="-1"
>
  <div>
    <div>
      TopBar Component
    </div>
    <div
      class="style__Wrapper-sc-184j6m9-0 iDVMYk"
    >
      <div
        class="StyleComponents__Container-sc-11erdxt-0 fFgwWO"
      >
        <div
          class="StyleComponents__Row-sc-11erdxt-1 jhoSVF"
        >
          <div
            class="style__SearchInput-sc-184j6m9-1 bWXmHs"
          >
            <input
              autocomplete="off"
              data-testid="search-input"
              name="search-input"
              placeholder="search by movie name!"
              type="text"
            />
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-search fa-w-16 searchIcon"
              data-icon="search"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                fill="currentColor"
              />
            </svg>
          </div>
          <ul
            id="movie-List"
          >
            <li>
              <a
                href="/premiered/movie-detail/5fa71823adc463397a50c502"
              >
                <div
                  class="image-wrapper"
                >
                  <img
                    alt=""
                    src="https://catalog.cinema-api.com/cf/images/ncg-images/68bfeef55ada4e7ab0606db2af0f40ad.jpg?width=240&version=EA2E80EE161E5DA161C6E26487054C9F&format=webp"
                  />
                </div>
                <div>
                  <p>
                    follow me
                  </p>
                  <p>
                    family
                    children
                  </p>
                  <p>
                    2020-10-05T15:18:52.722Z
                  </p>
                </div>
                <div
                  class="right-side"
                >
                  show
                </div>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Search Page Component Testing should render message when api returns empty list 1`] = `
<div
  style="outline: none;"
  tabindex="-1"
>
  <div>
    <div>
      TopBar Component
    </div>
    <div
      class="style__Wrapper-sc-184j6m9-0 iDVMYk"
    >
      <div
        class="StyleComponents__Container-sc-11erdxt-0 fFgwWO"
      >
        <div
          class="StyleComponents__Row-sc-11erdxt-1 jhoSVF"
        >
          <div
            class="style__SearchInput-sc-184j6m9-1 bWXmHs"
          >
            <input
              autocomplete="off"
              data-testid="search-input"
              name="search-input"
              placeholder="search by movie name!"
              type="text"
            />
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-search fa-w-16 searchIcon"
              data-icon="search"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                fill="currentColor"
              />
            </svg>
          </div>
          <ul
            id="movie-List"
          >
            <div
              class="no-result-message"
            >
              <p>
                Could not find movies with this search term, give me more characters.
              </p>
            </div>
            
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Search Page Component Testing take snapshot 1`] = `
<div
  style="outline: none;"
  tabindex="-1"
>
  <div>
    <div>
      TopBar Component
    </div>
    <div
      class="style__Wrapper-sc-184j6m9-0 iDVMYk"
    >
      <div
        class="StyleComponents__Container-sc-11erdxt-0 fFgwWO"
      >
        <div
          class="StyleComponents__Row-sc-11erdxt-1 jhoSVF"
        >
          <div
            class="style__SearchInput-sc-184j6m9-1 bWXmHs"
          >
            <input
              autocomplete="off"
              data-testid="search-input"
              name="search-input"
              placeholder="search by movie name!"
              type="text"
            />
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-search fa-w-16 searchIcon"
              data-icon="search"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                fill="currentColor"
              />
            </svg>
          </div>
          <ul
            id="movie-List"
          >
            
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
